samples/Common.c

```diff
@@ -880,14 +885,16 @@ STATUS freeSampleConfiguration(PSampleConfiguration* ppSampleConfiguration)
     pSampleConfiguration = *ppSampleConfiguration;
 
     CHK(pSampleConfiguration != NULL, retStatus);
-    for (i = 0; i < pSampleConfiguration->streamingSessionCount; ++i) {
+    ATOMIC_LOAD(&pSampleConfiguration->streamingSessionCount);
+    MUTEX_LOCK(pSampleConfiguration->sampleConfigurationObjLock);
```

Comment:
I actually think if we got here then the mutex is valid. However, think if you want to move the locks outside of the loop and in that case you will need to check whether the mutex is valid before executing the entire loop